core/src/main/java/edu/wpi/grip/core/util/LazyInit.java

```diff
@@ -0,0 +1,44 @@
+package edu.wpi.grip.core.util;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * A holder for data that gets lazily initialized.
+ *
+ * @param <T> the type of held data
+ */
+public class LazyInit<T> {
+
+  private T value = null;
+  private final Supplier<? extends T> factory;
+
+  /**
+   * Creates a new lazily initialized data holder.
+   *
+   * @param factory the factory to use to create the held value
+   */
+  public LazyInit(Supplier<? extends T> factory) {
```

Comment:
If you don't make this expose a constructor, it doesn't tightly couple this to needing to be newed.